Makefile

```diff
@@ -57,10 +61,36 @@ test: generate $(ALL_GO_MODS:%=test/%)
 test/%/go.mod:
 	@cd $* && $(GOCMD) test $(ARGS) ./...
 
-.PHONY: generate
-generate: export CFLAGS := $(BPF_INCLUDE)
-generate: go-mod-tidy
-generate:
+PROBE_ROOT = internal/pkg/instrumentation/bpf/
+PROBE_DIRS = $(shell find $(PROBE_ROOT) -type d -exec sh -c '(ls -p "{}"|grep />/dev/null)||dirname "{}"' \;)
+PROBE_GEN := $(addsuffix /bpf_x86_bpfel.go,$(PROBE_DIRS))
+PROBE_GEN += $(addsuffix /bpf_arm64_bpfel.go,$(PROBE_DIRS))
+PROBE_GEN := $(wildcard $(PROBE_GEN))
+PROBE_GEN += internal/pkg/instrumentation/bpf/net/http/client/bpf_no_tp_arm64_bpfel.go
+PROBE_GEN += internal/pkg/instrumentation/bpf/net/http/client/bpf_no_tp_x86_bpfel.go
```

Comment:
This is similar to other comments: If we add more compilation variants this will need to be updated.